src/pip/_internal/operations/install/wheel.py

```diff
@@ -595,9 +595,11 @@ def pyc_source_file_paths() -> Generator[str, None, None]:
         # Sorting installation paths makes it easier to reproduce and debug
         # issues related to permissions on existing files.
         for installed_path in sorted(set(installed.values())):
-            full_installed_path = os.path.join(lib_dir, installed_path)
+            full_installed_path = os.path.abspath(os.path.join(lib_dir, installed_path))
             if not os.path.isfile(full_installed_path):
                 continue
+            if lib_dir not in full_installed_path:
```

Comment:
Would it be possible to use `os.path.commonpath` or `commonprefix` here instead?